libmuscle/cpp/src/libmuscle/data.cpp

```diff
@@ -962,7 +962,7 @@ DataConstRef DataConstRef::grid_data_<bool>(
         Data result = Data::byte_array(num_elems);
         char * data_copy = result.as_byte_array();
         std::copy(data, data + num_elems, data_copy);
-        return result;
+        return std::move(result);
```

Comment:
Clang 14 doesn't give me any warnings for this. Doesn't this remove the possibility for RVO, like in the other cases that got flagged in the other direction?

Reply:
Ah, maybe it would be better then to avoid the split return, and create `result` before the branch, then assign according to what we need? That would then hopefully not give a warning on any clang version, and enable NRVO too.

Also, maybe I should extend the CI to run on the older Ubuntu versions as well, like with gcc, because especially clusters often have older environments, and the more compatible the better.

Reply:
> Ah, maybe it would be better then to avoid the split return, and create result before the branch, then assign according to what we need? That would then hopefully not give a warning on any clang version, and enable NRVO too.

Just tried that and the warning is still there on clang 10... Anyway a copy of Data is not very expensive so I'll leave it at this now.

Reply:
Funny, I added integration tests on Ubuntu 18.04 and 20.4, and clang 6 doesn't warn, clang 10 does, and clang 14 doesn't. Maybe 6 isn't smart enough to realise there's a problem, 10 can see that there is a problem but doesn't know how to solve it, and 14 is smart enough to do the RVO and then doesn't need to warn? Well, good enough I'd say. I'm going to merge this as is indeed.